drivers/comparator/comparator_stm32_comp.c

```diff
@@ -0,0 +1,324 @@
+/*
+ * Copyright (c) 2025 Alexander Kozhinov <ak.alexander.kozhinov@gmail.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/irq.h>
+#include <zephyr/kernel.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/util_macro.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/comparator.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+#include <zephyr/drivers/interrupt_controller/intc_exti_stm32.h>
+
+#include <stm32_ll_comp.h>
+#include <stm32_ll_system.h>
+
+LOG_MODULE_REGISTER(stm32_comp, CONFIG_COMPARATOR_LOG_LEVEL);
+
+#define DT_DRV_COMPAT st_stm32_comp
+
+#define LL_COMP_INPUT_PLUS_IN0		LL_COMP_INPUT_PLUS_IO1
+#define LL_COMP_INPUT_PLUS_IN1		LL_COMP_INPUT_PLUS_IO2
+
+#define LL_COMP_INPUT_MINUS_IN0		LL_COMP_INPUT_MINUS_IO1
+#define LL_COMP_INPUT_MINUS_IN1		LL_COMP_INPUT_MINUS_IO2
+
+#define STM32_COMP_DT_INST_P_IN(inst)	\
+	CONCAT(LL_COMP_INPUT_PLUS_, DT_INST_STRING_TOKEN(inst, positive_input))
+
+#define STM32_COMP_DT_INST_N_IN(inst)	\
+	CONCAT(LL_COMP_INPUT_MINUS_, DT_INST_STRING_TOKEN(inst, negative_input))
+
+#define STM32_COMP_DT_INST_HYST_MODE(inst)	\
+	CONCAT(LL_COMP_HYSTERESIS_, DT_INST_STRING_TOKEN(inst, hysteresis))
```

Comment:
`hysteresis` property seems optional according to DT bindings YAML file. It is required? Should there be a default value?